src/components/UI/Inputs/NumberPicker/index.tsx

```diff
@@ -187,7 +187,7 @@ const NumberPicker = React.forwardRef<HTMLInputElement, INumberPickerProps>(
     };
 
     useEffect(() => {
-      if (prevMax !== max || prevMin !== min) {
+      if ((prevMax && prevMax !== max) || (prevMin && prevMin !== min)) {
```

Comment:
This is to prevent `calling the callback` when the component is rendered for the first time